redhat-upgrade-tool.py

```diff
@@ -365,6 +368,17 @@ def handle_sigwinch(signum, frame):
         upgrade_boot_args()
         prep_boot(kernel, initrd)
 
+    # Check for available space in /boot/ needed for kernel and grub
+    # installation during the upgrade.
+    statvfs = os.statvfs("/boot")
+    avail_bytes = statvfs.f_frsize * statvfs.f_bavail
+    if avail_bytes < MIN_AVAIL_BYTES_FOR_BOOT:
+        reset_boot()
+        additional_mb_needed = (MIN_AVAIL_BYTES_FOR_BOOT - avail_bytes) / 1e6
+        sys.stderr.write("Not enough space. /boot/ needs additional %d MB.\n"
+                         % int(additional_mb_needed))
```

Comment:
Last, but not least: It's not conventional to use SI Mega prefix here. When talking about data, the common convention is "mebibytes", that is, conversion rate 1024 (which can be conveniently written as `2**20`, `2**30`, etc.) and prefix `Mi`.